File: 657-robot-return-to-origin/robot-return-to-origin.c
bool judgeCircle(char* moves) {
    int index = 0;
    int hor = 0;
    int ver = 0;

    while(*(moves + sizeof(char) * index) != '\0'){
        if(*(moves + sizeof(char) * index) == 'U'){
            ver++;
        }
        else if(*(moves + sizeof(char) * index) == 'D'){
            ver--;
        }
        else if(*(moves + sizeof(char) * index) == 'L'){
            hor--;
        }
        else if(*(moves + sizeof(char) * index) == 'R'){
            hor++;
        }

        index++;
    }

    return (hor == 0 && ver == 0) ? true : false;
}

File: 657-robot-return-to-origin/robot-return-to-origin.js
/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function(moves) {
    let horizontal = 0;
    let vertical = 0;
    for(let i = 0; i < moves.length; i++){
        if(moves[i] == 'U'){
            vertical++;
            continue;
        }

        if(moves[i] == 'D'){
            vertical--;
            continue;
        }

        if(moves[i] == 'L'){
            horizontal--;
            continue;
        }

        if(moves[i] == 'R'){
            horizontal++;
        }
    }

    return !horizontal && !vertical;
};